csrc/quantization/compressed_tensors/int8_quant_kernels.cu

```diff
@@ -103,134 +105,165 @@ static inline __device__ int8_t int32_to_int8(int32_t x) {
-  int const tid = threadIdx.x;
-  int64_t const token_idx = blockIdx.x;
-  scale_type const scale = *scale_ptr;
-
-  // Must be performed using 64-bit math to avoid integer overflow.
-  out += token_idx * hidden_size;
-  input += token_idx * hidden_size;
-
-  for (int i = tid; i < hidden_size; i += blockDim.x) {
-    out[i] = float_to_int8_rn(static_cast<float>(input[i]) / scale);
-  }
+    const scalar_t* __restrict__ input, int8_t* __restrict__ output,
+    const scale_t* scale_ptr, const int hidden_size) {
+  int tid = threadIdx.x;
+  int stride = blockDim.x;
+  float scale = *scale_ptr;
+
+  const scalar_t* row_in = input + blockIdx.x * hidden_size;
+  int8_t* row_out = output + blockIdx.x * hidden_size;
+
+  vectorize_with_alignment(
+      row_in, row_out, hidden_size, tid, stride,
+      [=] __device__(vec_n_t<int8_t, 16> & dst,
+                     const vec_n_t<scalar_t, 16>& src) {
+#pragma unroll
+        for (int k = 0; k < 16; ++k)
+          dst.val[k] = float_to_int8_rn(float(src.val[k]) / scale);
+      },
+      [=] __device__(int8_t& dst, const scalar_t& src) {
+        dst = float_to_int8_rn(float(src) / scale);
+      });
 }
 
-template <typename scalar_t, typename scale_type, typename azp_type>
+template <typename scalar_t, typename scale_t, typename azp_t>
 __global__ void static_scaled_int8_azp_quant_kernel(
-    scalar_t const* __restrict__ input, int8_t* __restrict__ out,
-    scale_type const* scale_ptr, azp_type const* azp_ptr,
-    const int hidden_size) {
-  int const tid = threadIdx.x;
-  int64_t const token_idx = blockIdx.x;
-  scale_type const scale = *scale_ptr;
-  azp_type const azp = *azp_ptr;
-
-  // Must be performed using 64-bit math to avoid integer overflow.
-  out += token_idx * hidden_size;
-  input += token_idx * hidden_size;
-
-  for (int i = tid; i < hidden_size; i += blockDim.x) {
-    auto const val = static_cast<float>(input[i]);
-    auto const quant_val = int32_to_int8(float_to_int32_rn(val / scale) + azp);
-    out[i] = quant_val;
-  }
+    const scalar_t* __restrict__ input, int8_t* __restrict__ output,
+    const scale_t* scale_ptr, const azp_t* azp_ptr, const int hidden_size) {
+  int tid = threadIdx.x;
+  int stride = blockDim.x;
+  float scale = *scale_ptr;
+  azp_t azp = *azp_ptr;
+  float inv_s = 1.0f / scale;
+
+  const scalar_t* row_in = input + blockIdx.x * hidden_size;
+  int8_t* row_out = output + blockIdx.x * hidden_size;
+
+  vectorize_with_alignment(
+      row_in, row_out, hidden_size, tid, stride,
+      [=] __device__(vec_n_t<int8_t, 16> & dst,
+                     const vec_n_t<scalar_t, 16>& src) {
+#pragma unroll
+        for (int k = 0; k < 16; ++k) {
+          float v = float(src.val[k]) * inv_s;
+          dst.val[k] = int32_to_int8(float_to_int32_rn(v) + azp);
+        }
+      },
+      [=] __device__(int8_t& dst, const scalar_t& src) {
+        float v = float(src) * inv_s;
+        dst = int32_to_int8(float_to_int32_rn(v) + azp);
+      });
```

Comment:
Could we reuse this part between the static and dynamic quant kernels? Or even better, consolidate with the quant utilities in `csrc/quantization/fused_kernels/quant_conversions.cuh`?

Reply:
I think after the usage of `DefaultVecOp`, the caller side now is much easier, please take a look